app/models/load_balancer_listener.rb

```diff
@@ -15,4 +15,28 @@ class LoadBalancerListener < ApplicationRecord
   has_many :resource_groups, -> { distinct }, :through => :load_balancer_pool_members
 
   virtual_total :total_vms, :vms, :uses => :vms
+
+  def load_balancer_port_range_str
+    range_to_port_range_str(load_balancer_port_range)
+  end
+
+  def instance_port_range_str
+    range_to_port_range_str(instance_port_range)
+  end
+
+  private
+
+  # Convert a Range object into the string representation usually seen for port
+  # ranges (specifically, "80" or "3200-3233")
+  def range_to_port_range_str(r)
+    if r.empty?
+      ""
+    elsif r.size == 1
+      r.first.to_s
+    else
+      # Range#last apparently ignores exclusiveness without an explicit
+      # parameter
+      "#{r.first}-#{r.last(1)[0]}"
```

Comment:
Right, my comment is specifically to the behavior of `Range#last`. It doesn't return the last element in the set of values in the range (which is what I would expect); instead it returns the upper bound value of the range without regard to whether or not the range is exclusive or inclusive.  

In my opinion, I don't care if its range exclusive or inclusive. My comment regarding postgres preferring exclusive ranges is because even when using a constructor like `int4range(80, 80, '[]')', a SELECT on that value will return '[80, 81)'.

This is the same value and it doesn't really matter how it's stored!  I just want to make it clear that when we use `range_to_port_range_str`, it handles ranges that are both inclusive and exclusive properly.